uint8 id
bool is_logic
bool logic
---
bool success
